api/post.go

```diff
@@ -129,7 +131,7 @@ func CreateValetPost(c *Context, post *model.Post) (*model.Post, *model.AppError
 	return rpost, nil
 }
 
-func CreatePost(c *Context, post *model.Post, doUpdateLastViewed bool) (*model.Post, *model.AppError) {
+func CreatePost(c *Context, post *model.Post, teamId string, doUpdateLastViewed bool) (*model.Post, *model.AppError) {
```

Comment:
Is it worth mocking a session so that we need to pass one less argument into a function? I think it's cleaner to just pass in the team ID